### PR TITLE
Fix the log methods xxxln to output the correct format

### PR DESCRIPTION
**Pull request by jimmy-zh**
Fix the log methods `xxxln(Fatalln, Errorln, Warningln, Infoln, Debugln)` to output the correct format.

### CODE REVIEW

**Comment by jimmy-zh** [comment]
The same fix for context log methods. @tamalsaha Could you please have a review?

File: log/log.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func Fatal(args ...interface{}) {
 }
 
 func Fatalln(args ...interface{}) {
-	glog.FatalDepth(1, args...)
+	glog.FatalDepth(1, fmt.Sprintln(args...))
```
Comment (tamalsaha) [inline log/log.go]: Why is this needed?

File: log/context.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func (l *Logger) Fatal(args ...interface{}) {
 }
 
 func (l *Logger) Fatalln(args ...interface{}) {
-	glog.FatalDepth(1, l.ctx, args)
+	glog.FatalDepth(1, l.ctx, fmt.Sprintln(args...))
```
Comment (tamalsaha) [inline log/context.go]: Instead of turning this into a string here, should we just append it?

```
	glog.FatalDepth(1, append([]interface{}{l.ctx}, args...)...)
```

**Comment by tamalsaha** [comment]
Thanks for submitting the pr. Please take a look at my comments.

File: log/log.go (hunk 1 above)
Comment (jimmy-zh) [reply log/log.go]: Correct. `glog.fataldepth(1,args...)` has ensured that the output ends with a `\n`. However, the semantics of the ***ln method is not just to append a newline, but always add spaces between the operands, otherwise `args` will be outputted continuously. 

Here's a case when I ran `appscode/swift`:

https://github.com/appscode/swift/blob/122036233eb2effdcaf0fb3e3fbf336c45386ae3/pkg/release/chart.go#L186
```
log.Infoln("Downloading", repo.ChartURL, "to", filePath)
```
```
I0109 11:39:33.479314       1 chart.go:186] Downloadinghttps://hub.tencentyun.com/charts/tencenthub/charts/airflow-0.9.1.tgzto/tmp/swift872379824/airflow-0.9.1.tgz
```

Another example in `grpc/grpc-go`:

https://github.com/grpc/grpc-go/blob/bd0b3b2aa2a9c87b323ee812359b0e9cda680dad/grpclog/glogger/glogger.go#L76

File: log/context.go (hunk 2 above)
Comment (jimmy-zh) [reply log/context.go]: Yes. `append` is a good way to handle `l.ctx`. Based on the above comments, my suggestion is:
```suggestion
        glog.FatalDepth(1, fmt.Sprintln(append([]interface{}{l.ctx}, args...)...))
```

**Comment by jimmy-zh** [comment]
@tamalsaha Thank you for your reply. I added some new comments. Please check them.

File: log/log.go (hunk 1 above)
Comment (tamalsaha) [reply log/log.go]:
```
However, the semantics of the ***ln method is not just to append a newline, but always add spaces between the operands, otherwise args will be outputted continuously.
```

I missed that part. Your pr makes sense now.

File: log/context.go (hunk 2 above)
Comment (tamalsaha) [reply log/context.go]: This sounds good.

**Comment by tamalsaha** [comment]
@jimmy-zh, please update the context.go file to use the append operation. Other things look good.

**Comment by jimmy-zh** [comment]
@tamalsaha I have updated that. Please take a look. Thx.
